reports/infrastructure-deployments/main.go

```diff
@@ -0,0 +1,198 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ministryofjustice/cloud-platform-environments/pkg/authenticate"
+	"github.com/ministryofjustice/cloud-platform-how-out-of-date-are-we/reports/pkg/hoodaw"
+	"github.com/shurcooL/githubv4"
+	"golang.org/x/oauth2"
+)
+
+// resourceMap is used to store both string:string and string:map[string]string key
+// value pairs. The HOODAW API requires the first entry of map to contain a string:string,
+// the rest of the map consists of a primary key (string) with a value containing a map (string:string)
+// of Kubernetes namespaces and hostnames that don't contain the annotation contained within
+// the variable 'annotation'.
+type resourceMap map[string]interface{}
+
+var (
+	hoodawApiKey   = flag.String("hoodawAPIKey", os.Getenv("HOODAW_API_KEY"), "API key to post data to the 'How out of date are we' API")
+	hoodawEndpoint = flag.String("hoodawEndpoint", "/infrastructure_deployments", "Endpoint to send the data to")
+	hoodawHost     = flag.String("hoodawHost", os.Getenv("HOODAW_HOST"), "Hostname of the 'How out of date are we' API")
+	org            = flag.String("org", "ministryofjustice", "GitHub user or organisation.")
+	repository     = flag.String("repository", "cloud-platform-infrastructure", "Repository to check the PR of.")
+	token          = flag.String("token", os.Getenv("GITHUB_OAUTH_TOKEN"), "Personal access token for GitHub API.")
+
+	endPoint  = *hoodawHost + *hoodawEndpoint
+	infraPath = "terraform/aws-accounts/cloud-platform-aws"
+
+	// Number of months to generate report
+	numMonths = 12
+	// number of PRs to fetch per month. set to 100 with assumption
+	// that there are no more than 100 prs in the infrastructure repo
+	prCount = 100
```

Comment:
I don't see `endpoint, infrapath, numMonths and prCount` as variables. These are static values and should be constants.

Reply:
Thanks @jasonBirchall . Updated. The endpoint is from 2 vars so left it as a var